godump.go

```diff
@@ -278,14 +294,19 @@ func printDumpHeader(out io.Writer, skip int) {
 var callerFn = runtime.Caller
 
 // findFirstNonInternalFrame iterates through the call stack to find the first non-internal frame.
-func findFirstNonInternalFrame() (string, int) {
-	for i := 2; i < 10; i++ {
+func findFirstNonInternalFrame(skip int) (string, int) {
+	for i := initialCallerSkip; i < defaultMaxStackDepth; i++ {
 		pc, file, line, ok := callerFn(i)
 		if !ok {
 			break
 		}
 		fn := runtime.FuncForPC(pc)
 		if fn == nil || !strings.Contains(fn.Name(), "godump") {
+			if skip > 0 {
+				skip--
+				continue
+			}
+
 			return file, line
 		}
```

Comment:
See https://github.com/goforj/godump/pull/31#discussion_r2181499821

```suggestion
		if fn == nil || !strings.Contains(fn.Name(), "godump.go") {
			if skip > 0 {
				skip--
				continue
			}

			return file, line
		}
```

Please note, you could find the current file name by using the 1st frame that is currently skipped by starting the loop at 2

I'm a bit worried the function might provide wrong result if someone provides a godump.go file in its project. But it's unlikely to happen, and if it happens it means the user would not have to use WithSkipStackFrames so it should be OK

Reply:
If it works the way I expected please add the tests I suggested here


https://github.com/goforj/godump/pull/31#discussion_r2179837071

Reply:
> If it works the way I expected please add the tests I suggested here
> 
> [#31 (comment)](https://github.com/goforj/godump/pull/31#discussion_r2179837071)

Done